forwarder/kernel-forwarder/pkg/kernelforwarder/local.go

```diff
@@ -56,26 +59,44 @@ func (k *KernelForwarder) createLocalConnection(crossConnect *crossconnect.Cross
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 
-	srcName := crossConnect.GetSource().GetMechanism().GetParameters()[common.InterfaceNameKey]
-	dstName := crossConnect.GetDestination().GetMechanism().GetParameters()[common.InterfaceNameKey]
+	rand.Seed(time.Now().UnixNano())
+
+	var err error
+	var tempName string
 	var srcNetNsInode string
 	var dstNetNsInode string
-	var err error
+	srcName := crossConnect.GetSource().GetMechanism().GetParameters()[common.InterfaceNameKey]
+	dstName := crossConnect.GetDestination().GetMechanism().GetParameters()[common.InterfaceNameKey]
+
+	/* In case source and destination names are the same, use temporary values during the setup of the VETH interfaces */
+	if dstName == srcName {
+		tempName = srcName
+		srcName = fmt.Sprintf("nsm%d", rand.Uint32())
+		dstName = fmt.Sprintf("nsm%d", rand.Uint32())
```

Comment:
Should we use here something like this?
```suggestion
		dstName = fmt.Sprintf("%s%d", dstName, rand.Uint32())
```

Reply:
This change is making the interface name to go over 15 characters and hits the linux interface max limit.

Reply:
Indeed, I've reverted it to the way it was at the beginning 👍